`v -autofree` error in Gitly

### Describe the bug

V throws errors with the `autofree` flag in the Gitly repository.

### Expected Behavior

Successful compilation.

### Current Behavior

```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/gitly.1747225759301539199.tmp.c:59934:63: error: extraneous ')' before ';'
                string* _t4 = (string*)(array_get_with_check(diff_parts, 2)));
                                                                            ^
/tmp/v_501/gitly.1747225759301539199.tmp.c:61909:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t5.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/gitly.1747225759301539199.tmp.c:61964:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t4.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/gitly.1747225759301539199.tmp.c:62009:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t4.data);
...
==================
```

### Reproduction Steps

Clone https://github.com/walkingdevel/gitly
Run `v -autofree .` in the `fix-git-push` branch.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

c766ce4

### Environment details (OS name and version, etc.)

 macOS 13.0

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -314,7 +314,7 @@ fn (mut g Gen) index_of_array(node ast.IndexExpr, sym ast.TypeSymbol) {
 				g.write('))')
 			}
 		}
-		if needs_clone {
+		if !gen_or && needs_clone {
 			g.write(')')
 		}
 		if gen_or {

